switchsdk/src/androidTest/java/net/gini/switchsdk/uitests/TakePictureScreenTest.java

```diff
@@ -0,0 +1,236 @@
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.is;
+
+import android.Manifest;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.SystemClock;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.espresso.NoMatchingViewException;
+import android.support.test.espresso.ViewAssertion;
+import android.support.test.espresso.ViewInteraction;
+import android.support.test.espresso.intent.rule.IntentsTestRule;
+import android.support.test.espresso.matcher.ViewMatchers;
+import android.support.test.filters.LargeTest;
+import android.support.test.rule.GrantPermissionRule;
+import android.support.test.runner.AndroidJUnit4;
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+
+import net.gini.switchsdk.OnboardingManager;
+import net.gini.switchsdk.ReviewPictureActivity;
+import net.gini.switchsdk.SwitchSdk;
+import net.gini.switchsdk.test.R;
+
+import org.hamcrest.core.IsInstanceOf;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(AndroidJUnit4.class)
+@LargeTest
+public class TakePictureScreenTest {
+
+    private static final int TIME_TO_WAIT_BETWEEN_STEPS = 1000;
+    private static final int TIME_TO_WAIT_FOR_ACTIVITY_TO_START = 4000;
+    @Rule
+    public IntentsTestRule<HostActivity> mActivityTestRule = new IntentsTestRule
+            <>(HostActivity.class, true, false);
+    @Rule
+    public GrantPermissionRule mRuntimePermissionRule = GrantPermissionRule.grant(
+            Manifest.permission.CAMERA);
+    private Context mTargetContext;
+
+    @Test
+    public void cameraScreen_rejectedPictureShouldNotAppearInDocumentBar() {
+
+        omitOnboarding();
+        startIntent();
+
+        SystemClock.sleep(TIME_TO_WAIT_BETWEEN_STEPS);
+        onView(withId(R.id.button_take_picture))
+                .perform(click());
+
+        SystemClock.sleep(TIME_TO_WAIT_FOR_ACTIVITY_TO_START);
+
+        onView(withId(R.id.button_discard)).perform(click());
+
+        SystemClock.sleep(1000);
+
+        onView(withId(R.id.image_overview)).check(new RecyclerViewItemCountAssertion(1));
+
+    }
+
+    @Test
+    public void cameraScreen_takePictureShouldOpenReviewScreen() {
+
+        omitOnboarding();
+        startIntent();
+
+        SystemClock.sleep(TIME_TO_WAIT_BETWEEN_STEPS);
+        onView(withId(R.id.button_take_picture))
+                .perform(click());
+
+        SystemClock.sleep(TIME_TO_WAIT_FOR_ACTIVITY_TO_START);
+        intended(hasComponent(ReviewPictureActivity.class.getName()));
+    }
+
+    @Test
+    public void firstLaunch_shouldShowOnboarding() {
+
+        startIntent();
+
+        SystemClock.sleep(3000); //TODO verify that animations on test device has been disabled
```

Comment:
One idea would be to have several constants like `TIME_TO_WAIT_BETWEEN_STEPS_FAST`, `TIME_TO_WAIT_BETWEEN_STEPS_MEDIUM` and `TIME_TO_WAIT_BETWEEN_STEPS_SLOW` to accommodate different situations - fast = default, medium = after a quick animation, slow = after a screen transition.